includes/blocks/class-wc-block-featured-product.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * Display the featured product block in the content.
+ *
+ * @package WooCommerce\Blocks
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Wrapper class for Featured Product callback.
+ */
+class WC_Block_Featured_Product {
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	protected static $block_name = 'featured-product';
+
+	/**
+	 * Default attribute values, should match what's set in JS `registerBlockType`.
+	 *
+	 * @var array
+	 */
+	protected static $defaults = array(
+		'align'        => 'none',
+		'contentAlign' => 'center',
+		'dimRatio'     => 50,
+		'showDesc'     => true,
+		'showPrice'    => true,
+	);
+
+	/**
+	 * Render the Featured Product block.
+	 *
+	 * @param array  $attributes Block attributes. Default empty array.
+	 * @param string $content    Block content. Default empty string.
+	 * @return string Rendered block type output.
+	 */
+	public static function render( $attributes, $content ) {
```

Comment:
I think we can leave that out for now, and if it's asked for we can think about adding something – especially since editing this render is not going to update the preview in the editor, so the preview in Gutenberg wouldn't match the frontend anymore.